examples/hpc/dla-driver/examples/validate_conv.rs

```diff
@@ -28,22 +26,17 @@ fn calculate_conv2d_out_param_dim(
 }
 
 fn validate_conv2d_tiny() -> bool {
-    let mut dla = Dla::new();
+    let dla = Dla::new();
+
+    sprintln!("din\r\n");
+    let mut din: Vec<i8> = uart_read_to_heap(75).into_iter().map(|x| x as i8).collect();
 
-    let mut din: Vec<i8> = vec![
-        0, 0, 0, 2, 0, 0, 1, 2, 1, 2, 0, 0, 1, 2, 0, 1, 0, 0, 0, 2, 0, 0, 1, 0, 1, 2, 0, 1, 0, 1,
-        0, 0, 2, 2, 1, 1, 0, 2, 1, 1, 2, 1, 2, 2, 1, 0, 0, 1, 1, 2, 0, 1, 1, 1, 0, 0, 2, 0, 1, 2,
-        1, 0, 0, 1, 2, 1, 1, 1, 0, 0, 1, 1, 2, 0, 2,
-    ];
-    let mut wgt: Vec<i8> = vec![
-        -1, -1, 0, -1, 0, 0, -1, -1, 1, 0, 0, 1, 1, -1, -1, 1, -1, 0, 1, 0, -1, -1, 1, -1, -1, 0,
-        -1, 1, 0, 0, -1, 0, 1, 0, -1, 1, 0, 1, -1, -1, 0, 0, 0, -1, -1, 0, -1, 1, -1, -1, -1, 0, 1,
-        0,
-    ];
-
-    let mut dout: Vec<i32> = vec![
-        -10, -1, -10, 0, -14, 2, -14, -4, -6, -5, -13, 4, -12, -2, -7, 1, -10, 0,
-    ];
+    sprintln!("wgt\r\n");
+    let mut wgt: Vec<i8> = uart_read_to_heap(54).into_iter().map(|x| x as i8).collect();
+
+    sprintln!("dout\r\n");
+    let dout: Vec<i8> = uart_read_to_heap(18).into_iter().map(|x| x as i8).collect();
+    let dout_i32: Vec<i32> = dout.into_iter().map(|x| x as i32).collect();
```

Comment:
This instructs an extra allocation for no particular reason. `.into_iter().map(i8::from).map(i32::from).collect()` is preferred, or even just `.into_iter().map(i32::from).collect()`.

Reply:
The conversion `map(i8::from)` doesn't seem to be equivalent to `map(|x| x as i8)`, since we want to cast the bits, not the value from u8. I removed the intermediary `dout` and cast `i8` to `i32` in the same pipeline.